ext/ddtrace_profiling_native_extension/private_vm_api_access.c

```diff
@@ -690,3 +690,28 @@ int ruby_thread_has_gvl_p(void) {
   return 0;
 }
 #endif // NO_THREAD_HAS_GVL
+
+#ifndef NO_RACTORS
+  // This API and definition are exported as a public symbol by the VM BUT the function header is not defined in any public header, so we
+  // repeat it here to be able to use in our code.
```

Comment:
Does it make sense to request/perform this change upstream?

Reply:
I think this "public-ish" is actually on purpose. That is, the missing header is not a "oops we forgot", it's a "this isn't really a public API, but some things outside of Ruby make good use of it, and we don't mind exposing it for now but you're on your own".

For instance in https://github.com/ruby/ruby/commit/15476c695d708796f2271d99deb3cd5288d016d2 you see a similar API that was published so that `fiddle` could use it:

https://github.com/ruby/ruby/blob/c3de7a3c58bf9a138ff8720ed56c0045d2b8e01d/internal/thread.h#L48

(Out of curiosity, there was some openness during a few chats with core team members recently to perhaps promote a few more APIs that would be really useful for profilers to this "public-ish" state).